hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java

```diff
@@ -325,22 +318,19 @@ private void resetUnconditionally(
 		}
 	}
 
-	private String reprOf(RichInstant consensusTime) {
-		return Optional.ofNullable(consensusTime)
-				.map(RichInstant::toJava)
-				.map(Object::toString)
-				.orElse("<N/A>");
+	private String reprOf(Instant consensusTime) {
+		return consensusTime == null ? "<N/A> " : consensusTime.toString();
```

Comment:
You have failed unit test for sure because of this typo (extra space at the end). We do have test for this. That is the power of testing every single line in the code.

Reply:
Tested it in local and it passed. I just re-ran the workflow of CircleCi, it's passed now. 
The failure is from this one: 

```
[ERROR] Process Exit Code: 134
[ERROR] Crashed tests:
[ERROR] com.hedera.services.ledger.HederLedgerTokensTest
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /repo/hedera-node && /usr/local/java/jdk-12.0.2/bin/java '-javaagent:/root/.m2/repository/org/jacoco/org.jacoco.agent/0.8.6/org.jacoco.agent-0.8.6-runtime.jar=destfile=/repo/hedera-node/target/jacoco.exec,excludes=com/hederahashgraph/**/*:com/hedera/services/proto/utils/**/*' -Xmx4096m -jar /repo/hedera-node/target/surefire/surefirebooter14075671043037827402.jar /repo/hedera-node/target/surefire 2021-05-26T02-01-51_134-jvmRun2 surefire1641739286551301472tmp surefire_11308172023960560073tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 134
[ERROR] Crashed tests:
[ERROR] com.hedera.services.ledger.HederLedgerTokensTest
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.awaitResultsDone(ForkStarter.java:510)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.runSuitesForkOnceMultiple(ForkStarter.java:382)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:297)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:246)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1183)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1011)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:857)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
```

Nevertheless, fixed the typo.

Reply:
hmm... I was deceived by the `<N/A>` here in this test. It was for the state version, not for the consensus time.

https://github.com/hashgraph/hedera-services/blob/e29caddc06e21724902867871a103e9b42d1d104/hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleNetworkContextTest.java#L191-L204

We should add a test for the `<N/A>` of consensus time. Otherwise later on we will still have to take care of it, since Codecov and SonarCloud will complain that the function `reprOf` was not fully covered in tests.

Reply:
Yeah, added one scenario for this.

Reply:
Thank you!